keras_cv/layers/preprocessing/cut_out.py

```diff
@@ -0,0 +1,148 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import tensorflow as tf
+import tensorflow.keras.layers as layers
+from tensorflow.keras import backend
+from tensorflow.python.keras.utils import layer_utils
+
+from keras_cv.utils import fill_utils
+
+
+class CutOut(layers.Layer):
+    """CutOut implements the CutOut data augmentation technique.
+
+    Args:
+        rate: Float between 0 and 1.  The fraction of samples to augment.
+        length: Integer. The side length of the square patches to cut out.
+        fill_mode: Pixels inside the patches are filled according
+            to the given mode (one of `{"constant", "gaussian_noise"}`).
+            - *constant*: Pixels are filled with the same constant value.
+            - *gaussian_noise*: Pixels are filled with random gaussian noise.
+        fill_value: a float represents the value to be filled inside the patches
+          when `fill_mode="constant"`.
+    References:
+       [CutOut paper](https://arxiv.org/abs/1708.04552).
+
+    Sample usage:
+    ```python
+    (images, labels), _ = tf.keras.datasets.cifar10.load_data()
+    cutout = keras_cv.layers.preprocessing.CutOut(1.0, 50)
+    augmented_images = cutout(images)
+    ```
+    """
+
+    def __init__(
+        self,
+        rate,
+        length,
+        fill_mode="constant",
+        fill_value=0.0,
+        seed=None,
+        name=None,
+        **kwargs
+    ):
+        super().__init__(name=name, **kwargs)
```

Comment:
You don't need to list `name` in the signature explicitly, you can just leave it as part of `kwargs`